docs/howto/krustlet-on-gke.md

```diff
@@ -0,0 +1,294 @@
+$ gcloud container clusters describe ${CLUSTER} --project=${PROJECT} --region=${REGION} --format="value(status)"
+RUNNING
+```
+
+Or:
+
+```shell
+$ kubectl get nodes
+NAME                                     STATUS   ROLES    AGE     VERSION
+gke-cluster-default-pool-1a3a5b85-scds   Ready    <none>   1m      v1.17.4-gke.10
+gke-cluster-default-pool-3885c0e3-6zw2   Ready    <none>   1m      v1.17.4-gke.10
+gke-cluster-default-pool-6d70a85d-19r8   Ready    <none>   1m      v1.17.4-gke.10
+```
+
+> **NOTE** If you chose to create a single-zone cluster, replace `--region=${REGION}` with `--zone=${ZONE}` in the above `gcloud` commands.
+
+## Step 1: Create a service account user for the node
+
+We will create a service account for Krustlet to use to register nodes and access specific secrets.
+
+This can be done by using the Kubernetes manifest in the [assets](./assets) directory:
+
+```shell
+$ kubectl apply --namespace=kube-system --filename=./docs/howto/assets/krustlet-service-account.yaml
+```
+
+Or by using the manifest directly from GitHub:
+
+```shell
+$ kubectl apply \
+--namespace=kube-system \
+--filename=https://raw.githubusercontent.com/deislabs/krustlet/master/docs/howto/assets/krustlet-service-account.yaml
+```
+
+Now that things are all set up, we need to generate the kubeconfig. You can do this by running
+(assuming you are in the root of the krustlet repo):
+
+```shell
+$ ./docs/howto/assets/generate-kubeconfig.sh
+```
+
+Or if you are feeling more trusting, you can run it straight from the repo:
+
+```shell
+bash <(curl https://raw.githubusercontent.com/deislabs/krustlet/master/docs/howto/assets/generate-kubeconfig.sh)
+```
+
+Either way, it will output a file called `kubeconfig-sa` in your current directory. Save this for
+later.
+
+## Step 1: Create Compute Engine VM
+
+We can create a new VM with the following command:
+
+```shell
+$ INSTANCE="krustlet" # Name of this VM must matches the certificate's CN
+$ # The cluster is distributed across the zones in the region
+$ # For the VM, we'll pick one of the zones
+$ ZONE="${REGION}-a" # Pick one of the zones in this region
+$ gcloud beta compute instances create ${INSTANCE} \
+--project=${PROJECT} \
+--zone=${ZONE} \
+--machine-type "n1-standard-1" \
+--image-family="debian-10" \
+--image-project="debian-cloud"
+NAME      ZONE        MACHINE_TYPE   PREEMPTIBLE  INTERNAL_IP  EXTERNAL_IP     STATUS
+instance  us-west1-a  n1-standard-1               xx.xx.xx.xx  yy.yy.yy.yy     RUNNING
+```
+
+Let's determine the instance's internal (!) IP to use when creating the Kubernete certificate and subsequently running Krustlet. In step #4, you'll need to copy this value into the command that is used to run Krustlet on the VM:
+
+```shell
+$ IP=$(gcloud compute instances describe ${INSTANCE} \
+--project=${PROJECT} \
+--zone=${ZONE} \
+--format="value(networkInterfaces[0].networkIP)") && echo ${IP}
+```
+
+It should take less than 30-seconds to provision the VM.
```

Comment:
Should this line be up around line 91? Feels out of place down here

Reply:
Fixed: 1775225d52484b2737d67129fb4ab5f320a963c5